src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1245,22 +1530,29 @@ public override BoundNode VisitLocalDeclaration(BoundLocalDeclaration node)
                 if (valueType.HasNullType)
                 {
                     Debug.Assert(type.IsErrorType());
-                    valueType = type.ToTypeWithState();
+
+                    if (!(initializer.Type is null))
+                    {
+                        valueType = type.ToTypeWithState();
+                    }
                 }
 
                 type = valueType.ToTypeSymbolWithAnnotations();
                 _variableTypes[local] = type;
             }
 
             TrackNullableStateForAssignment(initializer, type, slot, valueType, MakeSlot(initializer));
+            VisitTypeExpression(node.DeclaredType);
```

Comment:
>isitTypeExpression(node.DeclaredType); [](start = 13, length = 38)

Can this be moved up to the top of `VisitLocalDeclaration` and done unconditionally (whether initializer is null or not)?